lib/srv/app/watcher.go

```diff
@@ -97,6 +103,52 @@ func (s *Server) startResourceWatcher(ctx context.Context) (*services.AppWatcher
 	return watcher, nil
 }
 
+// guessPublicAddr will guess PublicAddr for given application if it is missing, based on proxy information and app name.
+func (s *Server) guessPublicAddr(app types.Application) types.Application {
```

Comment:
nit: How about naming this func `setPublicAddr`  instread of guess... /

Reply:
I chose `guess...` prefix to indicate we are using heuristic here, but if you think this can be misleading I can go back to `setPublicAddr`.